qiskit_machine_learning/runtime/torch_runtime_client.py

```diff
@@ -0,0 +1,513 @@
+        self._val_history = history
+
+    @property
+    def model_state_dict(self) -> Dict[str, Any]:
+        """Returns state dictionary of trained model"""
+        return self._model_state_dict
+
+    @model_state_dict.setter
+    def model_state_dict(self, state_dict: Dict[str, Any]) -> None:
+        """Sets state dictionary of trained model"""
+        self._model_state_dict = state_dict
+
+    @property
+    def train_time(self) -> float:
+        """Returns training time"""
+        return self._train_time
+
+    @train_time.setter
+    def train_time(self, time: float) -> None:
+        """Sets training time"""
+        self._train_time = time
+
+
+class TorchRuntimeClient:
+    """The Qiskit Machine Learning Torch runtime client to call the Torch runtime."""
+
+    def __init__(
+        self,
+        model: Module,
+        optimizer: Optimizer,
+        loss_func: Union[_Loss, Callable],
+        epochs: int = 10,
+        shots: int = 1024,
+        measurement_error_mitigation: bool = False,
+        provider: Optional[Provider] = None,
+        backend: Optional[Backend] = None,
+    ) -> None:
+        """
+        Args:
+            model: A PyTorch nn.Module to be trained.
+            optimizer: A PyTorch optimizer for the model parameters.
+            loss_func: A PyTorch-compatible loss function. Can be one of the
+                official PyTorch loss functions from ``torch.nn.loss`` or a custom
+                function defined by the user.
+            epochs: The maximum number of training epochs. By default, 10.
+            shots: The number of shots for the quantum backend. By default, 1024.
+            measurement_error_mitigation: Whether or not to use measurement error mitigation.
+            provider: IBMQ provider that supports runtime services.
+            backend: Selected quantum backend.
+        """
+        # Store settings
+        self._provider = None
+        self._model = None
+        self._loss_func = None
+        self._optimizer = None
+        self._epochs = epochs
+        self._shots = shots
+        self._measurement_error_mitigation = measurement_error_mitigation
+        self._backend = backend
+        self._score_func = None
+
+        # Use setter to check for valid inputs
+        if provider is not None:
+            self.provider = provider
+        self.optimizer = optimizer
+        self.loss_func = loss_func
+        self.model = model
+
+    @property
+    def provider(self) -> Optional[Provider]:
+        """Return the provider."""
+        return self._provider
+
+    @provider.setter
+    def provider(self, provider: Provider) -> None:
+        """Set the provider. Must be a provider that supports the runtime feature."""
+        try:
+            _ = hasattr(provider, "runtime")
+        except QiskitError:
+            # pylint: disable=raise-missing-from
+            raise ValueError(f"The provider {provider} does not provide a runtime environment.")
```

Comment:
I think it is better to add `ValueError` exception to the documentation instead of suppressing the warning.

Reply:
It was a lint issue. That's why I wrote the code. (I couldn't solve a lint error, but now I got it).